cpp/oneapi/dal/algo/kmeans/backend/gpu/train_kernel_lloyd_dense_dpc.cpp

```diff
@@ -155,18 +161,24 @@ struct train_kernel_gpu<Float, method::lloyd_dense, task::clustering> {
         auto updater = cluster_updater<Float>{ queue, comm }
                            .set_cluster_count(cluster_count)
                            .set_part_count(part_count)
-                           .set_data(arr_data);
+                           .set_data(arr_data)
+                           .set_data_squares(arr_data_squares);
         updater.allocate_buffers();
 
         for (iter = 0; iter < max_iteration_count; iter++) {
```

Comment:
why you don't declare iter here, i.e. std::int64_t iter and remove declaration on line 152, i found that you use it inside the loop only.

Reply:
I need it outside to report the total number of iterations